### ISSUE TITLE
Experiment Module: Add submit button to Proposals

### ISSUE THREAD

**Issue by aislinnpearson** [Rothamsted-Ecoinformatics/farm_rothamsted#658]
Please can we add a Submit button to the proposals. Exactly how we do this might need some thinking through... the ideal thing I think is a 'submit' button that appears next to the 'Save' button which is only present if the Proposal is in Draft. But maybe there are better ways of doing this... 

It would only appear for Research Leads. Farm Managers and Farm Data Admins.

**Comment by paul121** [comment]
> the ideal thing I think is a 'submit' button that appears next to the 'Save' button which is only present if the Proposal is in Draft. But maybe there are better ways of doing this...

Hey @aislinnpearson something to note is that the way these "Save" buttons (generally called "Action" buttons) have worked will be changing soon. They will be moving to the top of the page in all forms (including quick forms, settings pages, etc). Generally I think it is a good thing (better on long pages and small screens) but one side effect is that when there are multiple buttons, the 2nd, 3rd, etc, get moved into an expandable "..." menu.

It might be possible to get another big button next to the "primary action" button (often Save) but it would be a little hacky and I'd prefer to keep the standard behavior. What do you think? Does this affect this feature request of a "Submit" button?

See: https://github.com/farmOS/farmOS/pull/852

Example from an asset page:
![asset-more-actions](https://github.com/Rothamsted-Ecoinformatics/farm_rothamsted/assets/3116995/88de153e-f859-4a28-8cad-18e030a01e0b)

**Comment by paul121** [comment]
More generally, it sounds like we'd like to find an easy way to indicate a Proposal is in Draft state and could be Submitted with a single click. I wonder if we just need to make the `Status` easier to find in the form? Do you see how this Asset Status gets moved to the top of the sidebar - maybe this would help?

Similar to https://github.com/Rothamsted-Ecoinformatics/farm_rothamsted/issues/686, this sidebar is a relatively new thing that farmOS core assets + logs are using that we haven't adopted for research entities, but would be trivial/medium effort to adopt! The general theory we have is to put "metadata" in the sidebar. In farmOS core we want to avoid putting "important metadata" in the sidebar because it can be toggled/hidden on small screens, but I assume research data is often filled out on desktop?

For proposals I think it would make sense to move all of the "Status" tab into the sidebar. Maybe the following order: Status, Status notes, Reviewers, followed by Related experiments, designs, and plans?

This is a little off-topic from the original feature request - but curious what you think @aislinnpearson !

**Comment by paul121** [comment]
Another option could be to add an "Alert banner" whenever certain users are viewing a Draft Proposal (both the View + Edit pages). This could have a link to directly "Submit", or maybe a link to a small form to "Submit + add status/revision info". The alert could be Green or Yellow (same as you normally see). I think it would be effective at getting users' attention

A challenge if this is on the Edit page is that clicking the link would discard un-saved changes on the current page.

**Comment by aislinnpearson** [comment]
Hey @paul121,

Interesting... I love the save button at the top of the page though That is helpful.

> For proposals I think it would make sense to move all of the "Status" tab into the sidebar. Maybe the following order: Status, Status notes, Reviewers, followed by Related experiments, designs, and plans?

This is a good idea regardless. My only question would be are names/ fields for the related experiments, designs and plans too long to go in the sidebar? I also have a problem with those three fields where the scientists don't always understand the purpose them - ideally, they'd link to an existing experiment/ design _or_ request a new one, while the Study Plan would only be added once the proposal is approved.

In terms of where the submit button goes, I wonder if the other option is to keep the concepts of saving and submitting as separate and distinct concepts. So you save the form and then the submit button appears at the bottom of the screen (above the comments) or perhaps on the top left corner once you have created a first draft? Would something like that be easier to implement?

**Comment by aislinnpearson** [comment]
We agreed today that we would only include the submit button on the Proposal metadata screen (as opposed to the edit screen), partly to simplify this issue but also because this becomes complicated in terms of documenting edits _and_ then submitting. 

I have created a follow-up issue [here](https://github.com/Rothamsted-Ecoinformatics/farm_rothamsted/issues/709) to make sure the issue of what is included in the side menu is addressed separately.

**Comment by paul121** [comment]
@aislinnpearson here is what I've got. Let me know if you'd like any of this wording to be changed:

## 1. Display message with link to "submit proposal" when viewing the proposal
![Screenshot from 2024-07-22 19-29-08](https://github.com/user-attachments/assets/399bd9a6-1f6e-432d-bcc8-8948e06de101)

## 2. Clicking the link goes to a short, simple confirmation form with "Submit Proposal" button
![Screenshot from 2024-07-22 19-30-20](https://github.com/user-attachments/assets/44bd1906-3efd-40eb-b058-0a7278b20a6e)

## 3. After submission, redirect to the proposal view page and display message
![Screenshot from 2024-07-22 19-37-22](https://github.com/user-attachments/assets/8ecc8306-5196-4d8a-a0a9-e445bc24c6bc)

**Comment by paul121** [comment]
This looks good, lets change to a warning message.

**Comment by paul121** [comment]
Changed to a warning message

**Comment by aislinnpearson** [comment]
@paul121 this I think is good enough for a first version, but I don't know if it needs refining. We can test it with the users and see - I think a warning message has it's own benefits. I made a follow up issue here so we can keep track of it (#727 which is also linked to #728) 

My only question (and it's not important if we can't) is can we make the warning message red? If not, yellow is fine

**Comment by aislinnpearson** [comment]
Red is specifically an error message which suggests concern. Agreed to leave it yellow.
